src/FileService.Data/FileRepository.cs

```diff
@@ -2,46 +2,52 @@
 using LT.DigitalOffice.FileService.Data.Provider;
 using LT.DigitalOffice.FileService.Models.Db;
 using LT.DigitalOffice.Kernel.Exceptions.Models;
+using LT.DigitalOffice.Kernel.Extensions;
+using Microsoft.AspNetCore.Http;
 using System;
 using System.Linq;
 
 namespace LT.DigitalOffice.FileService.Data
 {
     public class FileRepository : IFileRepository
     {
-        private readonly IDataProvider provider;
+        private readonly IDataProvider _provider;
+        private readonly IHttpContextAccessor _httpContextAccessor;
 
-        public FileRepository(IDataProvider provider)
+        public FileRepository(IDataProvider provider, IHttpContextAccessor httpContextAccessor)
         {
-            this.provider = provider;
+            _provider = provider;
+            _httpContextAccessor = httpContextAccessor;
         }
 
         public Guid AddFile(DbFile file)
         {
-            provider.Files.Add(file);
-            provider.Save();
+            _provider.Files.Add(file);
+            _provider.Save();
 
             return file.Id;
         }
 
         public void DisableFile(Guid fileId)
         {
-            var dbFile = provider.Files.FirstOrDefault(file => file.Id == fileId);
+            var dbFile = _provider.Files.FirstOrDefault(file => file.Id == fileId);
 
             if (dbFile == null)
             {
                 throw new NotFoundException("File with this id was not found.");
             }
 
+            dbFile.ModifiedBy = _httpContextAccessor.HttpContext.GetUserId();
+            dbFile.ModifiedAtUtc = DateTime.UtcNow;
             dbFile.IsActive = false;
 
-            provider.Files.Update(dbFile);
-            provider.Save();
+            _provider.Files.Update(dbFile);
```

Comment:
do we need this? as far as I know, 33, 40-41 44 lines are enough for this logic

Reply:
done